src/services/collections.js

```
@@ -1,19 +1,79 @@
-import config from 'config';
+import { URL } from 'url';
 
 import fetch from '../utils/fetch.js';
+import logger from '../utils/logger.js';
 
-export const fetchCollections = async () => {
-  const collections = await fetch(config.get('@opentermsarchive/federation-api.collectionsUrl'));
+export async function fetchCollections(collectionsConfig) {
+  const errors = [];
+  const uniqueCollections = new Map();
 
-  return Object.keys(collections).reduce((result, collectionName) => {
-    if (collections[collectionName].endpoint) {
-      result.push({
-        name: collectionName,
-        id: collections[collectionName].id,
-        endpoint: collections[collectionName].endpoint,
+  (await Promise.allSettled(collectionsConfig.map(async item => {
+    let collections = [];
+
+    if (typeof item === 'string') {
+      collections = await fetch(item).catch(error => {
+        throw new Error(`${error.message} when fetching ${item}`);
       });
+      if (!Array.isArray(collections)) {
+        throw new Error(`Invalid format; an array of collections is expected when fetching ${item}`);
+      }
+    }
+
+    if (typeof item === 'object') {
+      collections = [item];
+    }
+
+    return filterInvalidCollections(collections);
```

In a promise approach, I'd find it clearer to only fetch in the first step, then chain the resulting array to `filter(validate)`, consider validation failures as rejected promises rather than log, and handle errors as a last step where we actually log (or throw).